### ISSUE TITLE
Implementation of neural network modules.

### ISSUE THREAD

**Issue by Napbad** [Napbad/Hahaha#41]
Implement the Sequential and related-Loss functions like `softmax`, `CrossEntropy`, etc. 
And create a simple public interface.
